common/src/main/java/org/opensearch/sql/common/utils/StringUtils.java

```diff
@@ -27,16 +27,28 @@ public static String unquote(String text, String mark) {
 
   /**
    * Unquote Identifier which has " or ' or ` as mark.
+   * Replace doubled quotes within the string with single version
    * @param text string
    * @return An unquoted string whose outer pair of (single/double/back-tick) quotes have been
    *     removed
    */
   public static String unquoteText(String text) {
-    if (isQuoted(text, "\"") || isQuoted(text, "'") || isQuoted(text, "`")) {
-      return text.substring(1, text.length() - 1);
-    } else {
+
+    if (isQuoted(text, "\"")) {
+      text = text.replace("\"\"", "\"");
+      text = text.substring(1, text.length() - 1).replace("\\\"", "\"").replace("\\\\", "\\");
+      return text;
+    }
+    if (isQuoted(text, "'")) {
+      text = text.replace("''", "'");
+      text = text.substring(1, text.length() - 1).replace("\\'", "'").replace("\\\\", "\\");
       return text;
```

Comment:
Strings like these: `""the cake" is lie"`, `''the cake' is lie'` would be converted into `the cake" is lie`/`the cake' is lie`. I think expected output there is `"the cake" lie`/`'the cake' is lie`.

We shouldn't collapse two `'`/`"` chars in the beginning and in the end into `'`/`"` char.
@MaxKsyunz, what do you think?